### DISCUSSION TITLE
How to return source document information with LCEL?

### DISCUSSION THREAD

**Discussion by navjotmakkar** [langchain-ai/langserve#458] [Q&A]
Hello, 

I am working on the example '[Conversational Retriever](https://github.com/langchain-ai/langserve/blob/main/examples/conversational_retrieval_chain/server.py)' in the readme of langserve repository.

How can i get the source document information passed and return to the response I get on calling the invoke api through langserve?

**Comment by eyurtsev** [comment]
What information do you want from the source document? Its contents or metadata?

**Reply by navjotmakkar** [reply]
I want the source document information so that i can provide where the response is referred from. In case of pdf, it can be like page number, section number or chapter. In case of website it can be the url, title or section or whatever. basically something that i can provide as a reference along with my response. Also, i was wondering since in case of retrieval process we usually get 3-4 chunk of docs, how would that work?

**Reply by shaojun** [reply]
actually you're asking for how to carry a variable (for you case, the source doc info) for all the chain, that means every runnable in the chain can access this variable. take a look at this : https://github.com/langchain-ai/langserve/discussions/479
